### PR TITLE
feat: split upload csv

### LINKED ISSUE #1560: Split /upload-csv endpoint
To simplify schema.

### PR DESCRIPTION
**Pull request by pyphilia**
close #1560

### CODE REVIEW

File: src/services/member/schemas.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ const currentAccountSchema = Type.Object(
 );
 
 // Current Member
-const compositeCurrentMemberSchema = Type.Composite([
```
Comment (pyphilia) [inline src/services/member/schemas.ts:72]: Remove.

File: src/services/item/plugins/invitation/schema.ts [hunk 2]
```diff
@@ -88,24 +88,28 @@ export const invite = {
 } as const satisfies FastifySchema;
 
 export const inviteFromCSV = {
+  params: entityIdSchemaRef,
+  response: {
+    [StatusCodes.OK]: Type.Object({
+      memberships: Type.Array(itemMembershipSchemaRef),
+      invitations: Type.Array(invitationSchemaRef),
+    }),
```
Comment (spaenleh) [inline src/services/item/plugins/invitation/schema.ts:96]: I think we are missing the potential error codes.

File: src/services/item/plugins/invitation/schema.ts [hunk 3]
```diff
@@ -88,24 +88,28 @@ export const invite = {
...
   response: {
-    [StatusCodes.OK]: Type.Union([
-      Type.Array(
-        Type.Object({
-          groupName: Type.String(),
-          memberships: Type.Array(itemMembershipSchemaRef),
-          invitations: Type.Array(invitationSchemaRef),
-        }),
-      ),
+    [StatusCodes.OK]: Type.Array(
```
Comment (spaenleh) [inline src/services/item/plugins/invitation/schema.ts:106]: Same here, I think we should add the potential error codes.

File: src/services/item/plugins/invitation/test/index.test.ts [hunk 4]
```diff
@@ -539,6 +552,118 @@ describe('Invitation Plugin', () => {
...
+        const response = await app.inject({
+          method: HttpMethod.Post,
+          url: `${ITEMS_ROUTE_PREFIX}/${item.id}/invitations/upload-csv-template`,
+          query: { templateId: templateItem.id },
+          payload: form,
+          headers: form.getHeaders(),
+        });
+        expect(response.statusCode).toEqual(StatusCodes.OK);
+        for (const group of response.json()) {
+          for (const inv of group.invitations) {
```
Comment (spaenleh) [inline src/services/item/plugins/invitation/test/index.test.ts:639]: You only test invitations, should we also test the memberships created ? Are there any in this test ?

**Review by spaenleh — CHANGES_REQUESTED** [review]
Looks good, thank you for the refactor and the tests, let me know about the comment.

**Review by spaenleh — APPROVED** [review]

**Pull request merged**